design/genome_attributes_filtering.md

```diff
@@ -0,0 +1,54 @@
+# Genome Attributes Filtering design v1
+
+* Add a meta endpoint that returns
+  * The column names
+  * The type for each column
+  * The filter strategy for types that have more than one
+  * Min and max range for numbers and dates
+  * Enum list for enums
+* Column types
+  * number - range filter
+    * Do we need to distinguish between int and float? We do for microtrait
+    * Allow specifying inclusivity at each end of the range
+  * date - range filter, ISO8601 input
+      * Allow specifying inclusivity at each end of the range
+  * string
+    * Maximum string length
+      * No need for this currently AFAWK
+    * Filter strategies
+      * prefix search
+      * full text search
+        * requires ArangoSearch View or inverted index
+      * substring search
+        * cannot use index without ArangoSearch View or (maybe?) inverted index
+          * TBD - will an inverted index work?
+        * Out of scope for v1
+  * Enum
```

Comment:
enum includes both list and dictionary type?

Reply:
Not sure what you mean - an enum is a set of strings

Reply:
I still have no idea what you mean - there aren't any dictionary values for enums

Reply:
I don't think there is a dictionary column type. Which column were you thinking of?

Reply:
currently I only see `_upas`.

Reply:
Those aren't visible to the user, they're for the service only

Reply:
https://github.com/kbase/collections/blob/main/src/service/data_products/genome_attributes.py#L212

Reply:
👍